src/DebuggerData.h

```diff
@@ -19,58 +19,62 @@ struct MemoryLayout
 	int mapperSize[4][4];
 };
 
+struct Breakpoint;
+
 class Breakpoints
 {
 public:
-	Breakpoints();
-
 	enum Type { BREAKPOINT = 0, WATCHPOINT_MEMREAD, WATCHPOINT_MEMWRITE,
 	            WATCHPOINT_IOREAD, WATCHPOINT_IOWRITE, CONDITION };
 
+	Breakpoints();
+
 	void clear();
 
 	void setMemoryLayout(MemoryLayout* ml);
 	void setBreakpoints(const QString& str);
 	QString mergeBreakpoints(const QString& str);
 	int breakpointCount();
-	bool isBreakpoint(quint16 addr, QString *id = nullptr);
-	bool isWatchpoint(quint16 addr, QString *id = nullptr);
+	bool isBreakpoint(quint16 addr, QString *id = nullptr, bool checkSlot = true);
+	bool isWatchpoint(quint16 addr, QString *id = nullptr, bool checkSlot = true);
+	quint16 getAddress(int bpIndex);
 
 	/* xml session file functions */
 	void saveBreakpoints(QXmlStreamWriter& xml);
 	void loadBreakpoints(QXmlStreamReader& xml);
 
 	int findBreakpoint(quint16 addr);
-	int findNextBreakpoint();
 
 	static QString createSetCommand(Type type, int address,
 	                                qint8 ps = -1, qint8 ss = -1, qint16 segment = -1,
 	                                int endRange = -1, QString condition = QString());
 	static QString createRemoveCommand(const QString& id);
 
+	const Breakpoint& getBreakpoint(int index);
+	bool inCurrentSlot(const Breakpoint& bp);
 private:
-	struct Breakpoint {
-		Type type;
-		QString id;
-		quint16 address;
-		// end for watchpoint region
-		quint16 regionEnd;
-		// gui specific condition variables
-		qint8 ps;
-		qint8 ss;
-		qint16 segment;
-		// general condition
-		QString condition;
-		// compare content
-		bool operator==(const Breakpoint &bp) const;
-	};
 
 	std::vector<Breakpoint> breakpoints;
-	MemoryLayout* memLayout;
+	MemoryLayout* memLayout = nullptr;
 
 	void parseCondition(Breakpoint& bp);
 	void insertBreakpoint(Breakpoint& bp);
-	bool inCurrentSlot(const Breakpoint& bp);
+};
+
+struct Breakpoint {
+	Breakpoints::Type type;
```

Comment:
Detail.
I wonder if it makes more sense to move the `Type` enum here (inside `Breakpoint`) instead of defining it in `Breakpoints`. I think 'type' is more a property of a single breakpoint rather than a property of a collection of breakpoints.